tango_simlib/utilities/fandango_json_parser.py

```diff
@@ -82,6 +82,12 @@ def preprocess_attribute_types(self, attribute_data):
                 if attr_prop == 'data_type':
                     attr_config[attr_prop] = getattr(CmdArgType, attr_prop_value)
                 elif attr_prop == 'data_format':
+                    # checking if SPECTRUM attr has max_dim_x key not registered
+                    if (attr_config[attr_prop] == 'SPECTRUM' and 
+                        'max_dim_x' not in attr_config.keys()):
+                        # include dim keys if condition is met
+                        attr_config.update(
+                            {'max_dim_x':len(attr_config['value']), 'max_dim_y': 0})
```

Comment:
We are in a loop iterating through `attr_config` items, so would prefer not to update that dict during the loop.  It works OK with Python 2.7 that creates a list of tuples from `items()` before starting the loop, but wouldn't work if we changed the `items()` to the generator version `iteritems()`.  In Python 3, `items()` has changed to the generator version.

Python 2.7:
```
In [4]: d = dict(a=1, b=2)

In [5]: for k, v in d.iteritems():
   ...:    if k == 'b':
   ...:        d.update({'c': 3})
   ...:         
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-f3edfcc5a2a0> in <module>()
----> 1 for k, v in d.iteritems():
      2     if k == 'b':
      3         d.update({'c': 3})
      4 

RuntimeError: dictionary changed size during iteration
```

Python 3:
```
In [7]: d = dict(a=1, b=2)

In [8]: for k, v in d.items():
   ...:     if k == 'b':
   ...:         d.update({'c': 3})
   ...:         
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-fb166d5c5dbc> in <module>()
----> 1 for k, v in d.items():
      2     if k == 'b':
      3         d.update({'c': 3})
      4 

RuntimeError: dictionary changed size during iteration
```

The way to work around this issue is to store all the updates you need in a temporary data structure.  After all the updates are known, use another loop iterate through this temporary data structure and apply the updates.   That way we never modify the sequence we are iterating over.